### PR TITLE
Feanil/gocd upgrade

### PR DESCRIPTION
**Pull request by feanil**
@edx/pipeline-team please review.

### CODE REVIEW

**Review by e0d — APPROVED** [review]
SGTM, was curious why the startup script/vars were being removed.

**Comment by feanil** [comment]
We added those when the ones provided by GoCD didn't work correctly.  They've since improved theirs and made changes that we would have to copy over wholesale if we wanted to keep those files up to date.

File: docker/build/go-agent/files/go-agent-start.sh [hunk 1]
```diff
@@ -1,23 +0,0 @@
...
-
-echo -e "${COLOR_START}Starting Go Agent to connect to server $GO_SERVER:$GO_SERVER_PORT ...${COLOR_END}"
-sed -i -e 's/GO_SERVER=.*/GO_SERVER='$GO_SERVER'/' /etc/default/go-agent
-sed -i -e 's/GO_SERVER_PORT=.*/GO_SERVER_PORT='$GO_SERVER_PORT'/' /etc/default/go-agent
-sed -i -e 's/GO_SERVER_SSL_PORT=.*/GO_SERVER_SSL_PORT='$GO_SERVER_SSL_PORT'/' /etc/default/go-agent
-
-mkdir -p /var/lib/go-agent/config
-/bin/rm -f /var/lib/go-agent/config/autoregister.properties
-
-AGENT_KEY="${AGENT_KEY:-123456789abcdef}"
```
Comment (macdiesel) [inline docker/build/go-agent/files/go-agent-start.sh:18]: Has the agent key stuff been done away with?

Comment (feanil) [reply docker/build/go-agent/files/go-agent-start.sh:18]: The upstream config file now does this: https://github.com/gocd/gocd-docker/blob/master/phusion/agent/go-agent-start.sh#L19

File: playbooks/roles/go-server/tasks/main.yml [hunk 2]
```diff
@@ -69,8 +69,7 @@
     mode: 0776
     owner: "{{ GO_SERVER_USER }}"
     group: "{{ GO_SERVER_GROUP }}"
-    # uncomment ansible has been upgraded to 2.0+
-    # checksum=md5:{{ item.md5 }}
+    checksum: "md5:{{ item.md5 }}"
```
Comment (doctoryes) [inline playbooks/roles/go-server/tasks/main.yml:72]: 👍

File: playbooks/roles/go-server/tasks/main.yml [hunk 3]
```diff
@@ -118,9 +117,9 @@
     regexp: "^{{ item.username }}"
     line: "{{ item.password_hash.stdout }}"
   with_items:
-    - { username: GO_SERVER_ADMIN_USERNAME, password: GO_SERVER_ADMIN_PASSWORD, password_hash: admin_user_password_line }
-    - { username: GO_SERVER_BACKUP_USERNAME, password: GO_SERVER_BACKUP_PASSWORD, password_hash: backup_user_password_line }
-    - { username: GO_SERVER_GOMATIC_USERNAME, password: GO_SERVER_GOMATIC_PASSWORD, password_hash: gomatic_user_password_line }
+    - { username: "{{ GO_SERVER_ADMIN_USERNAME }}", password: "{{ GO_SERVER_ADMIN_PASSWORD }}", password_hash: "{{ admin_user_password_line }}" }
+    - { username: "{{ GO_SERVER_BACKUP_USERNAME }}", password: "{{ GO_SERVER_BACKUP_PASSWORD }}", password_hash: "{{ backup_user_password_line }}" }
+    - { username: "{{ GO_SERVER_GOMATIC_USERNAME }}", password: "{{ GO_SERVER_GOMATIC_PASSWORD }}", password_hash: "{{ gomatic_user_password_line }}" }
```
Comment (doctoryes) [inline playbooks/roles/go-server/tasks/main.yml:122]: Wonder how this ever worked...

**Review by doctoryes — APPROVED** [review]

File: docker/build/go-agent/files/go-agent-start.sh (hunk 1 above)
Comment (macdiesel) [reply docker/build/go-agent/files/go-agent-start.sh:18]: outstanding

**Review by macdiesel — APPROVED** [review]

File: playbooks/roles/go-server/tasks/main.yml (hunk 3 above)
Comment (feanil) [reply playbooks/roles/go-server/tasks/main.yml:122]: It was fine in ansible 1.x but not in 2.x and our greps missed it.

**Pull request merged**